addons/web/static/src/model/relational_model/record.js

```diff
@@ -1049,20 +1085,21 @@ export class Record extends DataPoint {
                 }
             }
         }
-        if (Object.keys(changes).length > 0) {
-            const initialChanges = pick(this.data, ...Object.keys(changes));
-            this._applyChanges(changes);
+        const initialData = pick(
+            toRaw(this.data),
+            ...Object.keys(changes),
+            ...Object.keys(onchangeServerValues)
+        );
+        this._applyChanges(changes, onchangeServerValues);
+        if (Object.keys(changes).length > 0 || Object.keys(onchangeServerValues).length > 0) {
             try {
                 await this._onUpdate({ withoutParentUpdate });
             } catch (e) {
-                this._applyChanges(initialChanges);
+                this._applyChanges(initialData);
```

Comment:
in this case we may lose textvalues (false != "") information as we apply the initial data